pkg/components/metallb/component_test.go

```diff
@@ -15,9 +15,13 @@
 package metallb
 
 import (
+	"reflect"
 	"testing"
 
 	"github.com/hashicorp/hcl/v2"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
```

Comment:
I'd really like to avoid having these kind of dependencies. We don't really depend on the types there, in any way. It is just the way we are doing testing

What do you think of some alternative, like comparing the generated manifest string to the expected one and just compare strings? It will be sensitive to order or whitespace changes, though. But I think that is still a better trade off. 

Or is it there any other way we can do this?

Reply:
I think it's fine to have those dependencies. @rata any specific arguments why we shouldn't depend on them?

Reply:
We should be careful adding dependencies and, IMHO, we don't have any good reason to do it in this case. Something similar to what was mentioned here: https://github.com/kinvolk/lokomotive/pull/367#discussion_r419997419

But if this is something we are already doing, I think it is ok...

Reply:
Not sure if https://github.com/kinvolk/lokomotive/pull/367#discussion_r419997419 is a good example, as that case was about the single constant and here we use a complex type for decoding, which is more complicated to duplicate. We could still do that though.

Reply:
No, please don't! That is why I mentioned testing in another way, like just comparing strings! :joy: 

I just meant the same reasoning than there: not adding a dependency so lightly (in this case just for testing, not even use outside of the tests)

But, as I said, if we are doing this already, I think it is ok